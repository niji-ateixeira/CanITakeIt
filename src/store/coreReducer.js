// Ce fichier sert a regrouper les differant reducer a combin√©
// pour les ajouter dans le store par la suite
import { combineReducers } from 'redux';

import listDrugs from '../reducers/listDrugs';
// Liste des reducer a combiner pour les ajouter au store

// Combinaison des different store
export default combineReducers({ listDrugs });
