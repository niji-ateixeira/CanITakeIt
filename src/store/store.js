/*
  Fichier permettant la cr√©ation du store

  Date de modification: 28/08/2017
*/

import { createStore, applyMiddleware } from 'redux';
import createSagaMiddleware from 'redux-saga';

import CoreReducer from './coreReducer';
import getDrugs from '../sagas/getDrugs';

const sagaMiddleware = createSagaMiddleware();
// Creation du store
const store = createStore(CoreReducer, applyMiddleware(sagaMiddleware));

sagaMiddleware.run(getDrugs);

export default store;
