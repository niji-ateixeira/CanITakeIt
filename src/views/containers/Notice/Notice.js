import React from 'react';
import {
  Text,
  ScrollView,
  View,
} from 'react-native';

import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import Accordion from 'react-native-collapsible/Accordion';

import styles from './NoticeStyles';

import Loader from '../../components/Loader';
import Error from '../../components/Error';


class Notice extends React.Component {
  renderSectionTitle = section => (
    <Text>
      {section.title}
    </Text>
  )

  renderHeader = section => (
    <View>
      <Text style={styles.headerText}>
        {section.title}
      </Text>
    </View>
  )

  renderContent = section => (
    <View>
      <Text>
        {section.content}
      </Text>
    </View>
  )

  render() {
    const { listDrugs, load, response } = this.props;

    let render = (
      <ScrollView style={{ flex: 1 }}>
        <Text>
          Hmmm
        </Text>
      </ScrollView>
    );

    if (load) {
      render = <Loader />;
    } else if (response) {
      render = (
        <ScrollView style={{ flex: 1, backgroundColor: '#eefbf4' }}>
          <Accordion
            sections={listDrugs}
            renderSectionTitle={() => <React.Fragment />}
            renderHeader={(this.renderHeader)}
            renderContent={this.renderContent}
          />
        </ScrollView>
      );
    } else {
      render = <Error />;
    }
    return render;
  }
}

// Notice.propTypes = {
//   listDrugs: PropTypes.arrayOf(),
//   response: PropTypes.bool(),
//   load: PropTypes.bool(),
// };

Notice.defaultProps = {
  listDrugs: [],
  response: false,
  load: true,
};

const mapStateToProps = state => ({
  listDrugs: state.listDrugs.listDrugs,
  response: state.listDrugs.ok,
  load: state.listDrugs.fetched,
});
export default connect(mapStateToProps)(Notice);
