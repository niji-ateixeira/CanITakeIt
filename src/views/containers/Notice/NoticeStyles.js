import { StyleSheet } from 'react-native';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff',
    alignItems: 'center',
    justifyContent: 'center',
  },
  containerHeader:{
    justifyContent: 'center',
    backgroundColor:"#a4c9c2",
    padding: 10,
    borderRadius: 4,
    borderWidth: 0.5,
    borderColor: '#d6d7da',
  },
  textHeader:{
    fontSize: 20,
    fontWeight: 'bold',
  },
  containerContent:{
    padding: 10
  },
  textContent:{
    fontSize: 16,
  },
  name: {
    fontSize: 25,
    justifyContent: 'center',
    padding: 10,
    marginVertical: 20
  }
});

export default styles;
