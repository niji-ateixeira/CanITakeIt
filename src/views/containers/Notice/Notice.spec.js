import React from 'react';
import Notice from './Notice';

import renderer from 'react-test-renderer';

import { Provider } from 'react-redux';
import store from '../../../store/store';

test('renders correctly', () => {
  const tree = renderer.create(<Provider store={store}><Notice /></Provider>).toJSON();
  expect(tree).toMatchSnapshot();
});
