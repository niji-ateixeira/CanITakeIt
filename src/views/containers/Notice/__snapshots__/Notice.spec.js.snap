// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTScrollView
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View>
    <View />
  </View>
</RCTScrollView>
`;
